### PR TITLE
[DROOLS-6541] fix oopath following a not pattern

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6541

### CODE REVIEW

**Review by pibizza — APPROVED** [review]
Seems good to me

**Review by hellowdan — APPROVED** [review]

File: drools-core/src/main/java/org/drools/core/reteoo/ReactiveFromNodeLeftTuple.java [hunk 1]
```diff
@@ -45,9 +45,10 @@ public ReactiveFromNodeLeftTuple(final LeftTuple leftTuple,
                                      final boolean leftTupleMemoryEnabled) {
         super(leftTuple, rightTuple, currentLeftChild, currentRightChild, sink, leftTupleMemoryEnabled);
 
-        objects = new Object[leftTuple.getIndex() + 2];
-        System.arraycopy( leftTuple.toObjects(), 0, objects, 0, leftTuple.getIndex()+1 );
-        objects[leftTuple.getIndex()+1] = rightTuple.getFactHandle().getObject();
+        Object[] leftObjects = leftTuple.toObjects();
```
Comment (lucamolteni) [inline drools-core/src/main/java/org/drools/core/reteoo/ReactiveFromNodeLeftTuple.java]: why a +2 became a +1? Please explain these magic numbers by putting a comment

**Review by lucamolteni — CHANGES_REQUESTED** [review]

File: drools-core/src/main/java/org/drools/core/reteoo/ReactiveFromNodeLeftTuple.java (hunk 1 above)
Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/reteoo/ReactiveFromNodeLeftTuple.java]: @lucamolteni I found a further issue (another constructor wasn't applying the same fix), so I refactored and also added a comment to clarify it. Please review again.

**Review by lucamolteni — APPROVED** [review]
Ok thanks

**Pull request merged**